### PR TITLE
Fixes for Sigma RPC functions and HD Mints

### PR DESCRIPTION
**Pull request by riordant**
- Fixes issues with RPC tests failing for certain Sigma RPC functions
- Fixes the following error msg showing in debug.log:
```
ERROR: CheckZerocoinTransaction :  invalid denomination value, unable to convert to integer
```
- Fixes GUI out-of-sync issues with mints
- Fixes lag issue where a wallet with a large number of Sigma transactions is used
- Add reminted coins to HDMints
- Remove SIGMA_ERROR in place of boost::optional

### CODE REVIEW

**Comment by nopslide** [comment]
This pull request **introduces 1 alert** and **fixes 1** when merging 14a68c3ec3164a91fa2862f823e400ea830e795b into db3756cb08832f45032fec6c42b3700e1fa7ed6f - [view on LGTM.com](https://lgtm.com/projects/g/zcoinofficial/zcoin/rev/pr-d589211c3b15017fae00b0edd4d6c2aa199c83c8)

**new alerts:**

* 1 for &#39;import \*&#39; may pollute namespace

**fixed alerts:**

* 1 for Syntax error

---

This project has automated code review enabled, but doesn't use the <a href="https://github.com/apps/lgtm-com">LGTM GitHub App</a>. Migrate over by <a href="https://github.com/apps/lgtm-com/installations/new">installing the app</a>. Read about the <a href="https://lgtm.com/blog/announcing_github_apps_migration">benefits of migrating to GitHub Apps</a> in the blog.

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by reubenyap** [comment]
Does this also fix the intermittent lag issues?

File: src/sigma/coin.h [hunk 1]
```diff
@@ -19,7 +19,7 @@ enum class CoinDenomination : std::uint8_t {
     SIGMA_DENOM_10 = 3,
     SIGMA_DENOM_25 = 6,
     SIGMA_DENOM_100 = 4,
-    SIGMA_ERROR = 7
+    SIGMA_ERROR = 7
```
Comment (ultimaweapon) [inline src/sigma/coin.h]: This is not the right way to fix. The reason is we introduce a new possible value for denomination. The problems is not adding a new value but adding a new value that is not valid while any other values that not valid already invalid value.

Could you explain what you the problems you are trying to solve?

Comment (riordant) [reply src/sigma/coin.h]: Trying to introduce a default value for denomination, so when a `HDMint` object is initialized or set to NULL we reset the denomination value.

Comment (ultimaweapon) [reply src/sigma/coin.h]: Take a look on `boost::optional` (https://www.boost.org/doc/libs/1_65_1/libs/optional/doc/html/index.html). It should solve your problems without introduce a new special value for denomination.

**Comment by nopslide** [comment]
This pull request **introduces 1 alert** and **fixes 1** when merging b494a55bba540a0be1ec41a31251011cd313144c into 7b0f6a7f47e2f84cd6ba0b9761946374d63b635f - [view on LGTM.com](https://lgtm.com/projects/g/zcoinofficial/zcoin/rev/pr-d471e744cebf99823e6c9a43c6a2e202e4261d2b)

**new alerts:**

* 1 for &#39;import \*&#39; may pollute namespace

**fixed alerts:**

* 1 for Syntax error

---

This project has automated code review enabled, but doesn't use the <a href="https://github.com/apps/lgtm-com">LGTM GitHub App</a>. Migrate over by <a href="https://github.com/apps/lgtm-com/installations/new">installing the app</a>. Read about the <a href="https://lgtm.com/blog/announcing_github_apps_migration">benefits of migrating to GitHub Apps</a> in the blog.

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by nopslide** [comment]
This pull request **introduces 1 alert** and **fixes 1** when merging 624858bdd893a8d089ce294afa902ec43d1c3bdb into 7b0f6a7f47e2f84cd6ba0b9761946374d63b635f - [view on LGTM.com](https://lgtm.com/projects/g/zcoinofficial/zcoin/rev/pr-d02ed776c11036c64d050dc27162cc73dee6598b)

**new alerts:**

* 1 for &#39;import \*&#39; may pollute namespace

**fixed alerts:**

* 1 for Syntax error

---

This project has automated code review enabled, but doesn't use the <a href="https://github.com/apps/lgtm-com">LGTM GitHub App</a>. Migrate over by <a href="https://github.com/apps/lgtm-com/installations/new">installing the app</a>. Read about the <a href="https://lgtm.com/blog/announcing_github_apps_migration">benefits of migrating to GitHub Apps</a> in the blog.

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by nopslide** [comment]
This pull request **introduces 1 alert** and **fixes 1** when merging 97b755a1f92ae6f59793281a2e46c55f80db61de into 7b0f6a7f47e2f84cd6ba0b9761946374d63b635f - [view on LGTM.com](https://lgtm.com/projects/g/zcoinofficial/zcoin/rev/pr-4681a09c64040c1e0e940622eeb7f27f44ebd782)

**new alerts:**

* 1 for &#39;import \*&#39; may pollute namespace

**fixed alerts:**

* 1 for Syntax error

---

This project has automated code review enabled, but doesn't use the <a href="https://github.com/apps/lgtm-com">LGTM GitHub App</a>. Migrate over by <a href="https://github.com/apps/lgtm-com/installations/new">installing the app</a>. Read about the <a href="https://lgtm.com/blog/announcing_github_apps_migration">benefits of migrating to GitHub Apps</a> in the blog.

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Pull request merged**